### PR TITLE
Fix params for a controlled - UnitaryGate

### LINKED ISSUE #5750: UnitaryGate.control().inverse() increases the number of qubits in the instruction
### Information

- **Qiskit Terra version**: 0.14.1
- **Python version**: 3.7
- **Operating system**: Windows 10

### What is the current behavior?

Generating the inverse of a circuit using `circuit.inverse()` does not yield the correct circuit.

### Steps to reproduce the problem

A minimal example is this:
```
#%% Load packages

import numpy as np
from qiskit.tools.visualization import circuit_drawer
from qiskit.circuit import QuantumCircuit
from qiskit import execute, Aer


def calculate_unitary(circ, decimals=3):
    """ Calculate unitary matrix from circuit """
    backend = Aer.get_backend('unitary_simulator')
    job = execute(circ, backend)
    result = job.result()
    U=result.get_unitary(circ, decimals=decimals)
    return U

#%% Define a quantum circuit
def generate_full_phase_gate():        
    """ Generate a gate that acts by adding a minus sign to the entire state
    
    That is a bit funny gate, but it is valid and when combined to a conditional gate also a gate with physical meaning
    """
    multi = QuantumCircuit(1, name='P')
    multi.z(0)
    multi.x(0)
    multi.z(0)
    multi.x(0)
    full_phase_gate = multi.to_gate()
    return full_phase_gate

def generate_conditional_gate():
    qc = QuantumCircuit(3, name='conditional')
    
    p_gate = generate_full_phase_gate()
   
    cmulti_z_gate1 = p_gate.control(2, ctrl_state='01')
    qc.append(cmulti_z_gate1, [0,1, 2])

    return qc

encoder=generate_conditional_gate()
circuit_drawer(encoder, output='mpl')

#%% We generate a circuit by adding the conditional gate defined above, together with the inverse

qc = QuantumCircuit(3,3)
qc.extend(generate_conditional_gate())
qc.barrier()
qc.extend(generate_conditional_gate().inverse())
U=calculate_unitary(qc, decimals=16)
print(np.round(np.real_if_close(U), 14)+0) # should be the identity!
```

### What is the expected behavior?

Given any circuit, applying the circuit and then the inverse should yield an identity circuit. This is not the case for the minimal example.

### Suggested solutions

The bug seems to be triggered by the generation of the control gate in combination with the custom gate I generated. I tried generating this gate as 
```
from qiskit.extensions import UnitaryGate
full_phase_gate=UnitaryGate([[-1,0],[0,-1]])
```
But that gives similar problems. Perhaps the `UnitaryGate([[-1,0],[0,-1]])` is optimized to `UnitaryGate([[1,0],[0,1]])` in the calculation of the inverse?

### PR DESCRIPTION
**Pull request by georgios-ts**
### Summary
Resolves #5750.

### Details and comments
While building a controlled `UnitaryGate`, the `control_matrix` was given as a parameter in `ControlledGate`.  But `params` attribute of a `ControlledGate` corresponds to parameters of its `base_gate` (see [L199](https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/circuit/controlledgate.py#L199)). 
Effectively, it changed the matrix of the base `UnitaryGate` and`inverse` was building the inverse of this larger control matrix.

### CODE REVIEW

File: test/python/circuit/test_unitary.py [hunk 1]
```diff
@@ -296,3 +296,10 @@ def test_unitary_decomposition_via_definition_2q(self):
         """Test decomposition for 2Q unitary via definition."""
         mat = numpy.array([[0, 0, 1, 0], [0, 0, 0, -1], [1, 0, 0, 0], [0, -1, 0, 0]])
         numpy.allclose(Operator(UnitaryGate(mat).definition).data, mat)
+
+    def test_unitary_control(self):
+        """Test parameters of controlled - unitary."""
+        mat = numpy.array([[0, 1], [1, 0]])
+        gate = UnitaryGate(mat).control()
+        numpy.allclose(gate.params, mat)
```
Comment (Cryoris) [inline test/python/circuit/test_unitary.py:304]: Does this need to be in a `self.assertTrue` call to raise an error if it's not all close?

**Comment by Cryoris** [comment]
Thanks for the fix! Could you add a bugfix releasenote?

**Comment by ewinston** [comment]
Could you elaborate on the release note a bit so someone familiar with the effect would recognize its description? Thanks!

**Comment by Cryoris** [comment]
@georgios-ts are you still pursuing this? 🙂

**Comment by peendebak** [comment]
@Cryoris @georgios-ts Thanks, it fixes my bug report  #5750.

File: test/python/circuit/test_unitary.py (hunk 1 above)
Comment (Cryoris) [reply test/python/circuit/test_unitary.py:301]: @georgios-ts I think this is the only comment left then we can merge 🙂

**Review by ewinston — APPROVED** [review]

**Pull request merged**